base_layer/wallet_ffi/src/lib.rs

```diff
@@ -1872,6 +1897,43 @@ pub unsafe extern "C" fn wallet_test_complete_sent_transaction(
     }
 }
 
+/// This function checks to determine if a TariCompletedTransaction was originally a TariPendingOutboundTransaction
+///
+/// ## Arguments
+/// `wallet` - The TariWallet pointer
+/// `tx` - The TariCompletedTransaction
+/// `error_out` - Pointer to an int which will be modified to an error code should one occur, may not be null. Functions
+/// as an out parameter.
+///
+/// ## Returns
+/// `bool` - Returns if the transaction was originally sent from the wallet
+#[no_mangle]
+pub unsafe extern "C" fn wallet_check_originating_completed_transaction(
+    wallet: *mut TariWallet,
+    tx: *mut TariCompletedTransaction,
+    error_out: *mut c_int,
+) -> bool
+{
+    let mut error = 0;
+    ptr::swap(error_out, &mut error as *mut c_int);
+    if wallet.is_null() {
+        error = LibWalletError::from(InterfaceError::NullError("wallet".to_string())).code;
+        ptr::swap(error_out, &mut error as *mut c_int);
+        return false;
+    }
+    if tx.is_null() {
+        error = LibWalletError::from(InterfaceError::NullError("tx".to_string())).code;
+        ptr::swap(error_out, &mut error as *mut c_int);
+        return false;
+    }
+
+    if (*tx).source_public_key.to_hex() == (*wallet).comms.node_identity().public_key().clone().to_hex() {
```

Comment:
Shouldn't need to convert these to hex, the keys have a PartialEq implementation which means they can be directly compared for equality.

Reply:
Done